File: java/selectionSort.java

public class selectionSort {
    public static void selctionSorting(int arr[]){
        for (int i = 0; i < arr.length-1; i++) {
            int minValue = i;
            for (int j =i+1 ; j < arr.length; j++) {
                if(arr[minValue] > arr[j]){
                    minValue = j;
                }
            }
            int temp = arr[i];
            arr[i] = arr[minValue];
            arr[minValue] = temp;   
        }
    } 
    public static void printSelectionSorting(int arr[]){
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i]+" ");
            
        }
        System.out.println();
    }
    public static void main(String[] args) {
        int arr[] = {3,6,2,1,8,7,4,5,3,1};
        selctionSorting(arr);
        printSelectionSorting(arr);
    }

    
}

File: DSA_C++/Sorting/Selection_Sort.cpp
#include <iostream>
using namespace std;


void selectionSort(int arr[], int n) {
    for (int i = 0; i < n - 1; i++) {
        int minIndex = i;
        
        for (int j = i + 1; j < n; j++){
            
            if (arr[j] < arr[minIndex]){
                minIndex = j;
            }
        }
        if (minIndex != i) {
            int temp = arr[minIndex];
            arr[minIndex] = arr[i];
            arr[i] = temp;
        }
    }
}

int main() {
    int arr[] = {9,2,5,24,45,32,4,12,11,33,44,23,43,13};
    int n = sizeof(arr) / sizeof(arr[0]);
    
    cout << "Original array: ";
    for (int i = 0; i < n; ++i) {
        cout << arr[i] << " ";
    }
    cout << endl;
    
    selectionSort(arr, n);
    
    cout << "Sorted array: ";
    for (int i = 0; i < n; ++i) {
        cout << arr[i] << " ";
    }
    cout << endl;
    
    return 0;
}